flink/src/test/java/org/apache/iceberg/flink/actions/TestMigrateAction.java

```diff
@@ -0,0 +1,276 @@
+
+    URL url = new URL(location + File.separator + "test." + format.name());
+    File dataFile = new File(url.getPath());
+
+    List<Object[]> expected = Lists.newArrayList();
+    try (FileAppender<Record> fileAppender = genericAppenderFactory.newAppender(Files.localOutput(dataFile), format)) {
+      for (int i = 0; i < 10; i++) {
+        Record record = SimpleDataUtil.createRecord(i, "iceberg");
+        fileAppender.add(record);
+        expected.add(new Object[] {i, "iceberg"});
+      }
+    }
+
+    int migrateFileCount =
+        Actions.migrateHive2Iceberg(env, flinkHiveCatalog, SOURCE_HIVE_DB_NAME, SOURCE_HIVE_TABLE_NAME,
+            validationCatalog, baseNamespace, DATABASE, TABLE_NAME_UNPARTITIONED).execute();
+    Assert.assertEquals("Should produce the expected file count.", 1, migrateFileCount);
+
+    sql("USE CATALOG %s", catalogName);
+    sql("USE %s", DATABASE);
+    List<Object[]> list = sql("SELECT * FROM %s", TABLE_NAME_UNPARTITIONED);
+    Assert.assertEquals("Should produce the expected records count.", 10, list.size());
+    Assert.assertArrayEquals("Should produce the expected records.", expected.toArray(), list.toArray());
+  }
+
+  @Test
+  public void testMigratePartition() throws IOException, TException {
+    getTableEnv().getConfig().setSqlDialect(SqlDialect.HIVE);
+    sql("CREATE TABLE %s (id INT, data STRING) PARTITIONED BY (p STRING) STORED AS %s", SOURCE_HIVE_TABLE_NAME,
+        format.name());
+
+    Partition hivePartition = createHivePartition(format);
+    getTableEnv().getConfig().setSqlDialect(SqlDialect.DEFAULT);
+
+    metastoreClient.add_partition(hivePartition);
+
+    Partition partition = metastoreClient.getPartition(SOURCE_HIVE_DB_NAME, SOURCE_HIVE_TABLE_NAME, "p=iceberg");
+    String location = partition.getSd().getLocation();
+
+    Schema schema = new Schema(
+        Types.NestedField.required(1, "id", Types.IntegerType.get()),
+        Types.NestedField.optional(2, "data", Types.StringType.get()));
+    GenericAppenderFactory genericAppenderFactory = new GenericAppenderFactory(schema);
+
+    URL url = new URL(location + File.separator + "test." + format.name());
+    File dataFile = new File(url.getPath());
+
+    List<Object[]> expected = Lists.newArrayList();
+    try (FileAppender<Record> fileAppender = genericAppenderFactory.newAppender(Files.localOutput(dataFile), format)) {
+      for (int i = 0; i < 10; i++) {
+        Record record = SimpleDataUtil.createRecord(i, "iceberg" + i);
+        fileAppender.add(record);
+        expected.add(new Object[] {i, "iceberg" + i, "iceberg"});
+      }
+    }
+
+    int migrateFileCount =
+        Actions.migrateHive2Iceberg(env, flinkHiveCatalog, SOURCE_HIVE_DB_NAME, SOURCE_HIVE_TABLE_NAME,
+            validationCatalog, baseNamespace, DATABASE, TABLE_NAME_UNPARTITIONED).execute();
+    Assert.assertEquals("Should produce the expected file count.", 1, migrateFileCount);
```

Comment:
if we are going to create 1 manifest per partition, then we can also add an assertion on `allManifests`